fix(topology): checluster resource to have matching labels for catalog entity (#730)
janus-idp/backstage-plugins: plugins/topology/README.md

    app.openshift.io/vcs-ref: <GIT_REPO_BRANCH>
  ```

  > Note: If Red Hat OpenShift Dev Spaces is [installed and configured](https://access.redhat.com/documentation/en-us/red_hat_openshift_dev_spaces/3.7/html/administration_guide/installing-devspaces) and Git URL annotations are also present on the workload YAML file, then clicking on the edit code decorator redirects you to the Red Hat OpenShift Dev Spaces instance.

>   > Note: When you deploy the application using the OCP Git import flows, then you do not need to add the labels as import flows add the labels to the workload YAML file. Otherwise, you need to add the labels manually to the workload YAML file.

Reviewer: ```suggestion
When you deploy the Backstage application using the OCP git import flows, you do not need to add the labels as import flows. Otherwise, you need to add the labels to the workload YAML file manually.
```

Author: Instead of `Backstage` application I have used `your`. LMK what you think ?

Reviewer: I'd suggest explicitly stating the name of the application for the user

    app.openshift.io/vcs-ref: <GIT_REPO_BRANCH>
  ```

  > Note: If Red Hat OpenShift Dev Spaces is [installed and configured](https://access.redhat.com/documentation/en-us/red_hat_openshift_dev_spaces/3.7/html/administration_guide/installing-devspaces) and git URL annotations are also added to the workload YAML file, then clicking on the edit code decorator redirects you to the Red Hat OpenShift Dev Spaces instance.

>   > Note: When you deploy your application using the OCP git import flow, then you do not need to add the labels as import flow add the labels to the workload YAML file. Otherwise, you would need to add the labels to the workload YAML file manually.

Reviewer: ```suggestion
  > Note: When you deploy your application using the OCP git import flow, then you do not need to add the labels as import flow to the workload YAML file. Otherwise, you would need to add the labels to the workload YAML file manually.
```

  Other values result in icons not being rendered for the node.

- You can also add the `backstage.io/kubernetes-namespace` annotation to identify the Kubernetes resources using the defined namespace.

  ```yaml title="catalog-info.yaml"
  annotations:
    backstage.io/kubernetes-namespace: <RESOURCE_NS>
  ```

  > The Red Hat OpenShift Dev Spaces instance is not accessible using the edit code decorator if the `backstage.io/kubernetes-namespace` annotation is added to the `catalog-info.yaml` file.

  To retrieve the instance URL, CheCluster Custom Resource (CR) is required. The instance URL is not retrieved if the namespace annotation value is different from `openshift-devspaces` as CheCluster CR is created in `openshift-devspaces` namespace.

- A custom label selector is added, which Backstage uses to find the Kubernetes resources. The label selector takes precedence over the ID annotations.

  ```yaml title="catalog-info.yaml"
  annotations:
    backstage.io/kubernetes-label-selector: 'app=my-app,component=front-end'
  ```

>   If you have multiple entities while Red Hat Dev Spaces is configured and want multiple entities to be able to support the edit code decorator that redirects to the Red Hat Dev Spaces instance. You can add the `backstage.io/kubernetes-label-selector` annotation to the `catalog-info.yaml` file for each entity.

Reviewer 2: @debsmita1 above works as expected and does not conflict i.e. if `backstage.io/kubernetes-namespace` annotation is added to the `catalog-info.yaml` file then will the label selector work for entities across namespaces? 

or is it either?

Author: When the dev spaces is configured you must not add the namespace annotation to your entity file if you want to access it using the edit code decorator

<img width="1071" alt="Screenshot 2023-09-18 at 4 58 12 PM" src="https://github.com/janus-idp/backstage-plugins/assets/22490998/3726b62d-5f44-46d6-b7a2-e0ac56bc07f2">

If

  ```yaml title="catalog-info.yaml"
  annotations:
    backstage.io/kubernetes-label-selector: 'component in (<BACKSTAGE_ENTITY_NAME>,che)'
  ```

  If you are using the previous label selector, ensure that you add the following labels to your resources so that the Kubernetes plugin gets the Kubernetes resources from the requested entity:

  ```yaml
  labels: component=<BACKSTAGE_ENTITY_NAME>
  ```

>   ```yaml title="checluster.yaml"

Reviewer 2: what is `checluster.yaml`? do we need to provide some more context?

  labels: component=che
  ```

  You can also write your own custom query for the label selector with unique labels to differentiate your entities. However, you need to ensure that you add those labels to the resources associated with your entities including your CheCluster instance.

- The following label is added to the resources so that the Kubernetes plugin gets the Kubernetes resources from the requested entity:

  ```yaml
  labels:
    backstage.io/kubernetes-id: <BACKSTAGE_ENTITY_NAME>`

>   ```

Reviewer 2: this seems out of context to me, should this be with the annotations block or same https://github.com/janus-idp/backstage-plugins/pull/730/files#diff-53fc015c5ba5c543df0eb8f8fafb3604ef2b59b0c27c6d37ce3de17f8d0feeb7R158-R163

    backstage.io/kubernetes-id: <BACKSTAGE_ENTITY_NAME>
  ```

>   You can add the `backstage.io/kubernetes-label-selector` annotation if you have multiple entities and Red Hat OpenShift Dev Spaces is configured.

Reviewer: ```suggestion
If you have multiple entities and Red Hat Dev Spaces is configured, you can add the `backstage.io/kubernetes-label-selector` annotation to the `catalog-info.yaml` file. 
```

>   If you have multiple entities and Red Hat Dev Spaces is configured, you can add the `backstage.io/kubernetes-label-selector` annotation to the `catalog-info.yaml` file.

Reviewer 3: ```suggestion
  If you have multiple entities while Red Hat Dev Spaces is configured and want multiple entities to be able to support the edit code decorator that redirects to the Red Hat Dev Spaces instance. You can add the `backstage.io/kubernetes-label-selector` annotation to the `catalog-info.yaml` file.
```

Do you think it would make sense to specify the reason why they would want to use the `kubernetes-label-selector` annotation?

  ```yaml title="catalog-info.yaml"
  annotations:
    backstage.io/kubernetes-label-selector: 'component in (<BACKSTAGE_ENTITY_NAME>,che)'
  ```

>   If you are using the above label selector ensure that you add the following labels to your resources

Reviewer: ```suggestion
  If you are using the previous label selector ensure that you add the following labels to your resources:
```

>   ```yaml title="resource.yaml"

Reviewer 2: `resource.yaml` seems confusing to me, we can just say `The following label is added to the resources so that the Kubernetes plugin gets the Kubernetes resources from the requested entity`

>   If you are using the previous label selector ensure that you add the following labels to your resources so that the Kubernetes plugin gets the Kubernetes resources from the requested entity:

Reviewer 3: ```suggestion
  If you are using the previous label selector, ensure that you add the following labels to your resources so that the Kubernetes plugin gets the Kubernetes resources from the requested entity:
```

  labels: component=<BACKSTAGE_ENTITY_NAME>
  ```

  ```yaml title="devspaces.yaml"
  labels: component=che
  ```

>   Note: The `backstage.io/kubernetes-label-selector` annotation will override whatever you set for the `backstage.io/kubernetes-id` annotation.

Reviewer: ```suggestion
  **NOTE**: The `backstage.io/kubernetes-label-selector` annotation overrides the value you set for the `backstage.io/kubernetes-id` annotation.
```

>   ```yaml title="devspaces.yaml"

Reviewer 4: @debsmita1 to be more precise should we change this to `checluster.yaml` based on its kind?

>   You can even write your own custom query for the label selector with unique labels to differentiate your entities but ensure you add those labels to the resources associated with your entities including your CheCluster instance.

Reviewer: ```suggestion
  You can also write your own custom query for the label selector with unique labels to differentiate your entities. However, you need to ensure that you add those labels to the resources associated with your entities including your CheCluster instance.
```

- The following label is added to the resources so that the Kubernetes plugin gets the Kubernetes resources from the requested entity:

>   ```yaml title="resource.yaml"

Reviewer 2: This section coming below might confuse the user again, may we should have this before and at end add the details about label selector
